test/dotnet-watch.FunctionalTests/Scenario/ProjectToolScenario.cs

```diff
@@ -0,0 +1,150 @@
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Watcher.FunctionalTests
+{
+    public class ProjectToolScenario: IDisposable
+    {
+        private const string NugetConfigFileName = "NuGet.config";
+
+        public ProjectToolScenario()
+        {
+            Console.WriteLine($"The temporary test folder is {TempFolder}");
+
+            WorkFolder = Path.Combine(TempFolder, "work");
+            PackagesFolder = Path.Combine(TempFolder, "packages");
+
+            CreateTestDirectory();
+        }
+
+        public string TempFolder { get; } = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+
+        public string WorkFolder { get; }
+
+        public string PackagesFolder { get; }
+
+        public void AddProject(string projectFolder)
+        {
+            var destinationFolder = Path.Combine(WorkFolder, Path.GetFileName(projectFolder));
+            Console.WriteLine($"Copying project {projectFolder} to {destinationFolder}");
+
+            Directory.CreateDirectory(destinationFolder);
+
+            foreach (var directory in Directory.GetDirectories(projectFolder, "*", SearchOption.AllDirectories))
+            {
+                Directory.CreateDirectory(directory.Replace(projectFolder, destinationFolder));
+            }
+
+            foreach (var file in Directory.GetFiles(projectFolder, "*.*", SearchOption.AllDirectories))
+            {
+                File.Copy(file, file.Replace(projectFolder, destinationFolder), true);
+            }
+        }
+
+        public void AddNugetFeed(string feedName, string feed)
+        {
+            var tempNugetConfigFile = Path.Combine(WorkFolder, NugetConfigFileName);
+
+            var nugetConfig = XDocument.Load(tempNugetConfigFile);
+            var packageSource = nugetConfig.Element("configuration").Element("packageSources");
+            packageSource.Add(new XElement("add", new XAttribute("key", feedName), new XAttribute("value", feed)));
+            using (var stream = File.OpenWrite(tempNugetConfigFile))
+            {
+                nugetConfig.Save(stream);
+            }
+        }
+
+        public void AddToolToProject(string projectName, string toolName)
+        {
+            var projectFile = Path.Combine(WorkFolder, projectName, "project.json");
+            Console.WriteLine($"Adding {toolName} to {projectFile}");
+
+            var projectJson = JObject.Parse(File.ReadAllText(projectFile));
+            projectJson.Add("tools", new JObject(new JProperty(toolName, "1.0.0-*")));
+            File.WriteAllText(projectFile, projectJson.ToString());
+        }
+
+        public void Restore(string project = default(string))
+        {
+            if (project == default(string))
+            {
+                project = WorkFolder;
+            }
+            else
+            {
+                project = Path.Combine(WorkFolder, project);
+            }
+
+            var restore = ExecuteDotnet($"restore -v Minimal", project);
```

Comment:
How does this work? I don't think Minimal is a flag that's available on `dotnet`.